### PR TITLE
ROX-15458: Implement the back-channel from Central to Sensor for Node-Inventories 

### PR DESCRIPTION
**Pull request by vikin91**
## Description

Central (node-inventory pipeline) can report back to Sensor (nodeInventoryHandlerImpl) a successful scanning of the NodeInventory.

Left for the follow-up PR: Sensor should handle the received ACK and forward it to Compliance for handling.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added - more will be added in [ROX-16861](https://issues.redhat.com/browse/ROX-16861) 

### N/A
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

## Testing Performed

- Local deployment and looking at the logs
- CI

### CODE REVIEW

**Review by Maddosaurus — APPROVED** [review]
LGTM!
Only question I have: I vaguely remember we wanted to have the user-facing verbiage be "Node Scanning" instead of inventory. I'm personally okay with keeping the log messages in `pipeline.go` as they are, but maybe we should have them all consistent in their wording, right?

**Comment by vikin91** [comment]
> LGTM! Only question I have: I vaguely remember we wanted to have the user-facing verbiage be "Node Scanning" instead of inventory. I'm personally okay with keeping the log messages in `pipeline.go` as they are, but maybe we should have them all consistent in their wording, right?

Thanks! Yes, that is true. Actually the log messages here are more of us as for the customers, as the feature will work even without the ACKs. I will correct it on the next occasion - either in this PR or in the follow-up.

**Review by jschnath — APPROVED** [review]

File: proto/internalapi/central/sensor_iservice.proto [hunk 1]
```diff
@@ -65,6 +65,18 @@ message MsgToSensor {
...
+}
+
+message NodeInventoryACK {
+    string  cluster_id   = 1;
+    string  node_name = 2;
+    Action  action = 3;
+
+    enum Action {
+        ACK = 0;
+        NACK = 1;
```
[the remark this answers is not in the document]
Comment (vikin91) [reply proto/internalapi/central/sensor_iservice.proto:79]: This NACK is between Central and Sensor, the other NACK that you mention is between Sensor and Compliance. We need both of them.

**Comment by vikin91** [comment]
Thank you for the reviews @jschnath and @Maddosaurus !
I fixed failing unit tests and added more unit test coverage in  [d322655](https://github.com/stackrox/stackrox/pull/6155/commits/d322655234f7cada341ae4f6aa599c29d1be0eb5)

**Pull request merged**